tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraServicesDiscoveryRoutesContract.scala

```diff
@@ -0,0 +1,108 @@
+package com.linagora.tmail.james.common
+
+import io.netty.handler.codec.http.HttpHeaderNames.ACCEPT
+import io.restassured.RestAssured.{`given`, requestSpecification}
+import io.restassured.http.ContentType.JSON
+import net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson
+import org.apache.http.HttpStatus.{SC_OK, SC_UNAUTHORIZED}
+import org.apache.james.GuiceJamesServer
+import org.apache.james.jmap.http.UserCredential
+import org.apache.james.jmap.rfc8621.contract.Fixture.{ACCEPT_RFC8621_VERSION_HEADER, ANDRE, ANDRE_PASSWORD, BOB, BOB_PASSWORD, DOMAIN, authScheme, baseRequestSpecBuilder}
+import org.apache.james.utils.DataProbeImpl
+import org.hamcrest.Matchers.equalTo
+import org.junit.jupiter.api.{BeforeEach, Test}
+
+trait LinagoraServicesDiscoveryRoutesContract {
+  @BeforeEach
+  def setUp(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[DataProbeImpl])
+      .fluent()
+      .addDomain(DOMAIN.asString())
+      .addUser(BOB.asString(), BOB_PASSWORD)
+      .addUser(ANDRE.asString(), ANDRE_PASSWORD)
+
+    requestSpecification = baseRequestSpecBuilder(server)
+      .setAuth(authScheme(UserCredential(BOB, BOB_PASSWORD)))
+      .addHeader(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .build
+  }
+
+  @Test
+  def serviceDiscoveryShouldSucceedWhenUserIsAuthenticated(): Unit = {
+    val response: String = `given`()
+    .when()
+      .basePath(".well-known/linagora-ecosystem")
+      .get()
+    .`then`()
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response)
+      .isEqualTo(
+        s"""{
+           |	"linShareApiUrl": "https://linshare.linagora.com/linshare/webservice",
+           |	"linToApiUrl": "https://linto.ai/demo",
+           |	"linToApiKey": "apiKey",
```

Comment:
This call should likely be authenticated, as details like linto API key may be sensible...

Reply:
Yes, authentication has been done here. Test name `serviceDiscoveryShouldSucceedWhenUserIsAuthenticated` (basic auth in Before each BTW).

Maybe I would explicitly declare it for more readable.